common/components/org.wso2.carbon.data.agents.log4j/src/main/java/org/wso2/carbon/data/agents/log4j/appender/LogEventAppender.java

```diff
@@ -400,10 +397,9 @@ private void publishLogEvent(TenantAwareLoggingEvent event)
                 arbitraryDataMap.put(columns[8], stacktrace);
             }
             if (payLoadData != null && arbitraryDataMap != null) {
-                Event laEvent = new Event("loganalyzer:1.0.0", System.currentTimeMillis(), null, null,
+                Event laEvent = new Event(streamDef, System.currentTimeMillis(), null, null,
                         payLoadData.toArray(), arbitraryDataMap);
                 dataPublisher.publish(laEvent);
```

Comment:
If an exception occurs above within the try block, what will happen to this line (dataPublisher instance)? Will it get executed?